File: controle-tarefas-backend/src/main/java/com/example/controletarefas/dto/TarefaDto.java
package com.example.controletarefas.dto;

import com.example.controletarefas.model.Tarefa;
import jakarta.validation.constraints.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

public class TarefaDto {

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class Criacao {
        @NotBlank
        @Size(max = 100)
        private String nome;

        @Size(max = 500)
        private String descricao;

        @NotNull
        private Tarefa.Prioridade prioridade;

        @NotNull
        @FutureOrPresent
        private LocalDate dataPrevistaConclusao;
    }

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class Atualizacao {
        @NotBlank
        @Size(max = 100)
        private String nome;

        @Size(max = 500)
        private String descricao;

        @NotNull
        private Tarefa.Prioridade prioridade;

        @NotNull
        @FutureOrPresent
        private LocalDate dataPrevistaConclusao;
    }

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class Resposta {
        private UUID id;
        private String nome;
        private String descricao;
        private Tarefa.Prioridade prioridade;
        private Tarefa.Situacao situacao;
        private LocalDate dataPrevistaConclusao;
        private LocalDateTime dataCriacao;
    }

}

File: controle-tarefas-frontend/src/app/models/tarefa.dto.ts
import { Prioridade, Situacao } from './tarefa.enums';

export interface Tarefa {
  id?: string;
  nome: string;
  descricao?: string;
  prioridade: Prioridade;
  situacao: Situacao;
  dataPrevistaConclusao: string;
}

export interface TarefaResponse {
  id: string;
  nome: string;
  descricao?: string;
  prioridade: Prioridade;
  situacao: Situacao;
  dataPrevistaConclusao: string;
  dataCriacao: string;
}

export interface TarefaPage {
  content: TarefaResponse[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
}
